tests/test_url.py

```diff
@@ -418,6 +418,12 @@ def test_safe_url_string_encoding(
     # (%) are not escaped.
     f"a://example.com#{FRAGMENT_TO_ENCODE}",
 }
+if sys.version_info < (3, 11, 4):
+    KNOWN_SAFE_URL_STRING_URL_ISSUES.add("http://[2a01:5cc0:1:2:3:4]")  # Invalid IPv6
+else:
+    KNOWN_SAFE_URL_STRING_URL_ISSUES.add(
+        f"https://{USERNAME_TO_ENCODE}:{PASSWORD_TO_ENCODE}@example.com"
+    )
```

Comment:
So, to make sure I understand this correctly: 

1. Both `safe_url_string("http://[2a01:5cc0:1:2:3:4]")` and ```safe_url_string('https://"%;<=>@[]^`{|}\x7f:"%;<=>@[]^`{|}\x7f:@example.com')``` work on older Python versions but raise a ValueError on 3.11.4.
2. `test_safe_url_string_url` is a test that takes input and output of `safe_url_string()` from `SAFE_URL_URL_CASES` and expects that output unless the input is in `KNOWN_SAFE_URL_STRING_URL_ISSUES`.
3. Hence, `KNOWN_SAFE_URL_STRING_URL_ISSUES` is basically a list of cases that should xfail.
4. `SAFE_URL_URL_CASES` says that `"http://[2a01:5cc0:1:2:3:4]"` should give ValueError and this is true only on the new Python, so this change removes xfail for it on the new Python.
5. `SAFE_URL_URL_CASES` says that ```safe_url_string('https://"%;<=>@[]^`{|}\x7f:"%;<=>@[]^`{|}\x7f:@example.com')``` should be encoded correctly and that works only on older Python, so this change adds xfail for it on the new Python.

With this in mind I feel like we should instead change the USERNAME_TO_ENCODE and/or PASSWORD_TO_ENCODE so that it works on all Python versions instead.  It's probably enough to remove `[]` from there?

Reply:
Yes, if remove `[]` from `USERNAME_TO_ENCODE`, then test pass

Reply:
Cool, can you please make this change?